src/main/java/service/strategy/strategyimpl/PurchaseOperation.java

```diff
@@ -0,0 +1,20 @@
+package service.strategy.strategyimpl;
+
+import dao.Storage;
+import model.FruitRecord;
+
+
+public class PurchaseOperation implements OperationHandler {
+
+    @Override
+    public void apply(FruitRecord transaction) {
+        Integer currentQuantity = Storage.storage.get(transaction.getFruit());
+        int purchaseQuantity = transaction.getQuantity();
+        if (currentQuantity == null && currentQuantity < 0) {
+            throw new IllegalArgumentException("Invalid quantity of for purchase.");
+        }
+        int newQuantity = currentQuantity - purchaseQuantity;
```

Comment:
You need to check if the new quantity is positive before saving it to the `Storage`. If the new quantity is negative, throw a `RuntimeException` according to the checklist item 'Handling Purchase operation'.